clippy_lints/src/escape.rs

```diff
@@ -160,29 +160,29 @@ impl<'a, 'tcx: 'a> Delegate<'tcx> for EscapeDelegate<'a, 'tcx> {
 
         if let Categorization::Local(lid) = cmt.cat {
             if self.set.contains(&lid) {
-                if let Some(&Adjust::DerefRef { autoderefs, .. }) =
+                if let Some(&Adjust::Deref(ref overloaded)) =
                     self.tables
                         .adjustments
                         .get(&borrow_id)
                         .map(|a| &a.kind) {
                     if LoanCause::AutoRef == loan_cause {
                         // x.foo()
-                        if autoderefs == 0 {
+                        if overloaded == 0 {
                             self.set.remove(&lid); // Used without autodereffing (i.e. x.clone())
                         }
                     } else {
                         span_bug!(cmt.span, "Unknown adjusted AutoRef");
                     }
                 } else if LoanCause::AddrOf == loan_cause {
                     // &x
-                    if let Some(&Adjust::DerefRef { autoderefs, .. }) =
+                    if let Some(&Adjust::Deref(ref overloaded)) =
                         self.tables
                             .adjustments
                             .get(&self.tcx
                                 .hir
                                 .get_parent_node(borrow_id))
                             .map(|a| &a.kind) {
-                        if autoderefs <= 1 {
+                        if overloaded <= 1 {
                             // foo(&x) where no extra autoreffing is happening
```

Comment:
This check is weird, since it doesn't check for an autoref. Either way, it wants at most one deref adjustment.